web/modules/custom/customhook/customhook.module

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @file
+ * This file demonstrates  how to use hook function
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+
+/**
+ * Implements hook_cron()
+ */
+function customhook_cron() {
+  \Drupal::messenger()->addMessage(t('This message is from cron hook'));
+}
+
+/**
+ * Implements hook_form_alter()
+ */
+function customhook_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
+  if ($form_id == 'node_article_edit_form') {
+    $form['actions']['submit']['#value'] = t('Save your work');
+  }
+}
+
+/**
+ * Implements custom_hook_incremented
+ * This functions displays a message when a node is viewed for the first time
+ * @param int $current_count
+ * @param \Drupal\node\NodeInterface $node
+ *
+ */
+function customhook_incremented($current_count, \Drupal\node\NodeInterface $node) {
+  if ($current_count == 1) {
+    \Drupal::messenger()->addMessage(t('This is the first time you have viewed the
+     node %title. ', ['%title' => $node->label()]));
+  }
+}
+
+/**
+ * This function is implemented to display the number of times a node is visited
+ * in a particular session
+ *
+ * @param array $build
+ * @param EntityInterface $entity
+ * @return array
+ *
+ */
+function customhook_node_view(array &$build, EntityInterface $entity, EntityViewDisplay $display, $view_mode) {
+  $session = \Drupal::request()->getSession();
+  $current_counts = $session->get('customhook.view_counts', array());
+  if (!isset($current_counts[$entity->id()])) {
+    $current_counts[$entity->id()] = 1;
+  } else {
+    $current_counts[$entity->id()]++;
+  }
+  $session->set('customhook.view_counts', $current_counts);
+  $module_handler = \Drupal::moduleHandler();
+  $module_handler->invokeAll('incremented',
+  array($current_counts[$entity->id()], $entity));
+  $build['view_count'] = array(
+    '#markup' => '<p>' . t('You have viewed this node @total times this session.',
+    array('@total' => $current_counts[$entity->id()])) . '</p>',
+    '#cache' => array(
+      'max-age' => 0,
```

Comment:
Why max age = 0?

Reply:
In this hook i am displaying the number of time a node is being visited by the user, so the count must be updated on each page load. So for that scenario cache must be always be invalidated. That's `max age = 0`

Reply:
Ok. Good.

Reply:
You can resolve this comment. 
I am not seeing the button to be able to resolve the comment.

Reply:
Ok